isy-konfiguration/src/main/java/de/bund/bva/isyfact/konfiguration/common/impl/ReloadablePropertyKonfiguration.java

```diff
@@ -141,31 +116,25 @@ public synchronized boolean checkAndUpdate() {
     }
 
     /**
-     * Informiere alle registrieten Listener. Die geänderten Schlüssel werden aus den angegebenen Properties
+     * Informiere alle registrierten Listener. Die geänderten Schlüssel werden aus den angegebenen Properties
      * ermittelt.
      * @param bisherigeProperties
      *            Die bisherige Konfiguration.
      * @param neueProperties
      *            Die neue Konfiguration.
      */
     private void fireKonfigurationChanged(Properties bisherigeProperties, Properties neueProperties) {
-        Set<String> geaenderteSchluessel = ermittleGeaenderteSchluessel(bisherigeProperties, neueProperties);
+        Set<String> geaenderteSchluessel =
+            unmodifiableSet(ermittleGeaenderteSchluessel(bisherigeProperties, neueProperties));
         for (String schluessel : geaenderteSchluessel) {
             LOG.debug("Konfigurationsparameter {} wurde geändert.", schluessel);
         }
-        if (geaenderteSchluessel.size() > 0) {
-            List<KonfigurationChangeListener> notifyChangeListeners =
-                new ArrayList<>(this.konfigurationChangeListener.size());
+        if (!geaenderteSchluessel.isEmpty()) {
             synchronized (this.konfigurationChangeListener) {
-                if (this.konfigurationChangeListener.size() > 0) {
-                    // Verwende eine lokale Kopie, so dass sich Listener registrieren oder deregistrieren
-                    // können, wenn sich die Konfiguration geändert hat.
-                    notifyChangeListeners.addAll(this.konfigurationChangeListener);
```

Comment:
Angenommen es sollten die Listener in eine lokale Kopie, damit gleichzeitig "notified" und Listener hinzugefügt/entfernt werden könnten, dann hat das eh nicht geklappt, weil die ganze Logik im synchronized Block war...